lib/plugins/wallet/geth/base.js

```diff
@@ -63,28 +65,88 @@ function sendCoins (account, tx, settings, operatorId) {
 }
 
 function balance (account, cryptoCode, settings, operatorId) {
   return checkCryptoCode(cryptoCode)
-    .then(() => confirmedBalance(defaultAddress(account)))
+    .then(code => confirmedBalance(defaultAddress(account), code))
 }
 
-const pendingBalance = address => {
-  const promises = [_balance(true, address), _balance(false, address)]
+const pendingBalance = (address, cryptoCode) => {
+  const promises = [_balance(true, address, cryptoCode), _balance(false, address, cryptoCode)]
   return Promise.all(promises).then(([pending, confirmed]) => pending - confirmed)
 }
-const confirmedBalance = address => _balance(false, address)
+const confirmedBalance = (address, cryptoCode) => _balance(false, address, cryptoCode)
 
-function _balance (includePending, address) {
+function _balance (includePending, address, cryptoCode) {
+  if (coins.utils.getCryptoCurrency(cryptoCode).type === 'erc-20') {
+    const contract = web3.eth.contract(ABI.ERC20).at(coins.utils.getErc20Token(cryptoCode).contractAddress)
+    return contract.balanceOf(address.toLowerCase())
+  }
   const block = includePending ? 'pending' : undefined
-
   return pify(web3.eth.getBalance)(address.toLowerCase(), block)
 }
 
-function generateTx (_toAddress, wallet, amount, includesFee) {
+function generateTx (toAddress, wallet, amount, includesFee, cryptoCode) {
+  return coins.utils.getCryptoCurrency(cryptoCode).type === 'erc-20'
+    ? generateContractTx(toAddress, wallet, amount, includesFee, cryptoCode)
+    : generateCoinTx(toAddress, wallet, amount, includesFee)
+}
+
+function generateContractTx (_toAddress, wallet, amount, includesFee, cryptoCode) {
+  const contract = web3.eth.contract(ABI.ERC20).at(coins.utils.getErc20Token(cryptoCode).contractAddress)
+
+  const fromAddress = '0x' + wallet.getAddress().toString('hex')
+  const toAddress = _toAddress.toLowerCase()
+
+  const txTemplate = {
+    from: fromAddress,
+    to: coins.utils.getErc20Token(cryptoCode).contractAddress,
+    value: amount
+  }
+
+  const promises = [
+    pify(web3.eth.estimateGas)(txTemplate),
+    pify(web3.eth.getGasPrice)(),
+    pify(web3.eth.getTransactionCount)(fromAddress)
+  ]
+
+  return Promise.all(promises)
+    .then(arr => {
+      const gas = arr[0]
+      const gasPrice = arr[1]
+      const txCount = arr[2] <= lastUsedNonces[fromAddress]
+        ? lastUsedNonces[fromAddress] + 1
+        : arr[2]
+
+      const toSend = includesFee
+        ? amount.minus(gasPrice.times(gas))
+        : amount
+
+      const rawTx = {
+        nonce: txCount,
+        gasPrice: hex(gasPrice),
+        gasLimit: gas,
+        to: coins.utils.getErc20Token(cryptoCode).contractAddress,
+        from: fromAddress,
+        value: hex(BN(0)),
+        data: contract.transfer.getData(toAddress, hex(toSend))
+      }
+
+      const tx = new Tx(rawTx)
+      const privateKey = wallet.getPrivateKey()
+
+      tx.sign(privateKey)
+
+      return '0x' + tx.serialize().toString('hex')
+    })
```

Comment:
Most of this is duplicated code, can't we split off in a couple of functions?

Reply:
I think splitting off in multiple functions could get confusing fast, so I think I came with a more elegant solution. Will need some changes if we decide on more token standards, but for now I think it works well